src/Neo/Cryptography/Crypto.cs

```diff
@@ -50,22 +51,30 @@ public static byte[] Hash256(ReadOnlySpan<byte> message)
         }
 
         /// <summary>
-        /// Signs the specified message using the ECDSA algorithm.
+        /// Signs the specified message using the ECDSA algorithm and specified hash algorithm.
         /// </summary>
         /// <param name="message">The message to be signed.</param>
         /// <param name="priKey">The private key to be used.</param>
         /// <param name="ecCurve">The <see cref="ECC.ECCurve"/> curve of the signature, default is <see cref="ECC.ECCurve.Secp256r1"/>.</param>
+        /// <param name="hasher">The hash algorithm to hash the message, default is SHA256.</param>
         /// <returns>The ECDSA signature for the specified message.</returns>
-        public static byte[] Sign(byte[] message, byte[] priKey, ECC.ECCurve ecCurve = null)
+        public static byte[] Sign(byte[] message, byte[] priKey, ECC.ECCurve ecCurve = null, Hasher hasher = Hasher.SHA256)
         {
-            if (IsOSX && ecCurve == ECC.ECCurve.Secp256k1)
+            if (hasher == Hasher.Keccak256 || (IsOSX && ecCurve == ECC.ECCurve.Secp256k1))
```

Comment:
```suggestion
            if (hasher != Hasher.SHA256 || (IsOSX && ecCurve == ECC.ECCurve.Secp256k1))
```

also here

Reply:
is not possible to use None as `HashAlgorithmName` and always compute by ourself the hash?

Reply:
We need two different hashes to be supported. Also, unusual combinations like Secp256R1/Keccak and Koblitz/SHA256 are possible, thus I think hasher should be specified explicitly.

Reply:
I mean, always compute the hash by ourself, and set None as `HashAlgorithmName` to the native verify argument, is that possible?

Reply:
If you mean doing it like https://pkg.go.dev/crypto/ecdsa#Verify (which accepts _hash_ of the message and not the message itself) then it'd require an additional method, current `verifyWithECDsa` operates with `message` and I'd not change that behavior depending on `curve`/`curveHash`.